packages/packages/core/editor-global-classes/src/components/css-class-usage/components/css-class-usage-popover.tsx

```diff
@@ -0,0 +1,160 @@
+
+const iconMapper: Record< ContentType, { label: string; icon: React.ReactElement } > = {
+	'wp-post': {
+		label: __( 'Post', 'elementor' ),
+		icon: <PostTypeIcon fontSize={ 'inherit' } />,
+	},
+	'wp-page': {
+		label: __( 'Page', 'elementor' ),
+		icon: <PagesIcon fontSize={ 'inherit' } />,
+	},
+	popup: {
+		label: __( 'Popup', 'elementor' ),
+		icon: <PopupTemplateIcon fontSize={ 'inherit' } />,
+	},
+	header: {
+		label: __( 'Header', 'elementor' ),
+		icon: <HeaderTemplateIcon fontSize={ 'inherit' } />,
+	},
+	footer: {
+		label: __( 'Footer', 'elementor' ),
+		icon: <FooterTemplateIcon fontSize={ 'inherit' } />,
+	},
+};
+
+export const CssClassUsagePopover = ( {
+	cssClassID,
+	onClose,
+}: {
+	onClose: React.ComponentProps< typeof PopoverHeader >[ 'onClose' ];
+	cssClassID: string;
+} ) => {
+	const { data: classUsage } = useCssClassUsageByID( cssClassID );
+	const onNavigate = useOpenDocumentInNewTab();
+
+	const items: VirtualItem[] =
+		classUsage?.content.map(
+			( { title, elements, pageId, type } ) =>
+				( {
+					type: 'item',
+					value: pageId,
+					label: title,
+					secondaryText: elements.length.toString(),
+					docType: type,
+				} ) as VirtualItem
+		) ?? [];
+
+	return (
+		<>
+			<PopoverHeader
+				icon={ <CurrentLocationIcon fontSize={ 'tiny' } /> }
+				title={
+					<Stack flexDirection={ 'row' } gap={ 1 } alignItems={ 'center' }>
+						<Box aria-label={ 'header-title' }>{ __( 'Locator', 'elementor' ) }</Box>
+						<Box>
+							<Chip sx={ { lineHeight: 1 } } size={ 'tiny' } label={ classUsage.total } />
+						</Box>
+					</Stack>
+				}
+				onClose={ onClose }
+			/>
+			<Divider />
+			<PopoverBody width={ 300 }>
+				<PopoverMenuList
+					onSelect={ ( value ) => onNavigate( +value ) }
+					items={ items }
+					onClose={ () => {} }
+					menuListTemplate={ StyledCssClassUsageItem }
+					menuItemContentTemplate={ ( item ) => (
+						<Stack flexDirection={ 'row' } flex={ 1 } alignItems={ 'center' }>
+							<Box display={ 'flex' } sx={ { pr: 1 } }>
+								<Tooltip title={ iconMapper[ item.docType as ContentType ].label } placement="top">
+									<Icon co fontSize={ 'small' }>
+										{ iconMapper[ item.docType as ContentType ].icon }
+									</Icon>
+								</Tooltip>
+							</Box>
+							<Box sx={ { pr: 0.5, maxWidth: '173px' } } display={ 'flex' }>
+								<EllipsisWithTooltip
+									title={ item.label }
+									as={ Typography }
+									variant="caption"
+									maxWidth="173px"
+									sx={ {
+										lineHeight: 1,
+									} }
+								/>
+							</Box>
+							<ExternalLinkIcon className={ 'hover-only-icon' } fontSize={ 'tiny' } />
+							<Chip
+								sx={ {
+									ml: 'auto',
+								} }
+								size={ 'tiny' }
+								label={ item.secondaryText }
+							/>
+						</Stack>
+					) }
+				/>
+			</PopoverBody>
+		</>
+	);
+};
+
+const StyledCssClassUsageItem = styled( MenuList )( ( { theme } ) => ( {
```

Comment:
Talk with @IshayMaya: Are these our best practices? We try to avoid creating custom styles, am I wrong?